0-common/igluctl/src/main/scala/com.snowplowanalytics.iglu/ctl/GenerateCommand.scala

```diff
@@ -119,7 +122,65 @@ case class GenerateCommand(
       outputPair.map(_._1),
       migrations,
       outputPair.flatMap(_._2),
-      warnings = schemaErrors ++ ddlErrors ++ ddlWarnings)
+      warnings = schemaVerWarnings ++ schemaVerErrors ++ schemaErrors ++ ddlErrors ++ ddlWarnings)
+  }
+
+
+  /**
+    * Checks if there is any missing schema version in a directory of schemas
+    * or if a specific schema file doesn't have version 1-0-0
+    *
+    * @param schemas list of valid JSON Schemas including all Self-describing information
+    * @return (versionWarnings, versionErrors)
+    */
+  private[ctl] def validateSchemaVersions(schemas: List[IgluSchema]): (List[String], List[String]) = {
+
+    def existMissingSchemaVersion(schemaMaps: List[SchemaMap]): Boolean = {
+      val numOfMaps = schemaMaps.length
+
+      if (numOfMaps == 1){
+        false
+      } else {
+        val prevModel    = schemaMaps.head.version.model
+        val prevRevision = schemaMaps.head.version.revision
+        val prevAddition = schemaMaps.head.version.addition
+        val curModel     = schemaMaps.tail.head.version.model
+        val curRevision  = schemaMaps.tail.head.version.revision
+        val curAddition  = schemaMaps.tail.head.version.addition
+
+        if (curModel == prevModel && curRevision == prevRevision && curAddition == prevAddition + 1 ||
+            curModel == prevModel && curRevision == prevRevision + 1 && curAddition == 0 ||
+            curModel == prevModel + 1 && curRevision == 0 && curAddition == 0)
+          existMissingSchemaVersion(schemaMaps.tail) else true
+      }
+    }
+
+    if (schemas.empty) {
+      (List.empty[String], List.empty[String])
```

Comment:
Feel free to use `private case class MessageAccumulator(warnings: List[String], errors: List[String])`